src/ext/resources/legacygraphqlapi/schema.graphqls

```diff
@@ -2049,6 +2169,18 @@ type BookingInfo {
     dropOffMessage: String
 }
 
+input OptimizationCriteria @oneOf {
+    type: OptimizationType
+    triangle: InputTriangle
+}
+
+enum OptimizationType {
+    QUICK
+    SAFE
+    FLAT
+    GREENWAYS
```

Comment:
`GREENWAYS` is a strange feature that we should probably remove. I have no idea if it does what it says.

Reply:
I think we actually use it in our UI but I haven't really investigated what these different optimization types really did in a long time.

Reply:
We can discuss these optimization types in dev meeting